src/native/public/monoapi.cmake

```diff
@@ -47,25 +58,68 @@ set(metadata_public_headers_base
 	sgen-bridge.h
 	threads.h
 	tokentype.h
-	verify.h)
+	verify.h
+	details/environment-functions.h
+	details/opcodes-types.h
+	details/opcodes-functions.h
+	details/sgen-bridge-types.h
+	details/sgen-bridge-functions.h
+	details/image-types.h
+	details/image-functions.h
+	details/metadata-types.h
+	details/metadata-functions.h
+	details/assembly-types.h
+	details/assembly-functions.h
+	details/loader-types.h
+	details/loader-functions.h
+	details/class-types.h
+	details/class-functions.h
+	details/object-types.h
+	details/object-functions.h
+	details/exception-types.h
+	details/exception-functions.h
+	details/reflection-types.h
+	details/reflection-functions.h
+	details/appdomain-types.h
+	details/appdomain-functions.h
+	details/threads-types.h
+	details/threads-functions.h
+	details/debug-helpers-types.h
+	details/debug-helpers-functions.h
+	details/mono-debug-types.h
+	details/mono-debug-functions.h
+	details/mono-gc-types.h
+	details/mono-gc-functions.h
+	details/mono-config-types.h
+	details/mono-config-functions.h
+	details/profiler-types.h
+	details/profiler-functions.h
+	details/mono-private-unstable-types.h
+	details/mono-private-unstable-functions.h
+	)
 addprefix(metadata_public_headers ./mono/metadata/ "${metadata_public_headers_base}")
 
 set_target_properties(monoapi_metadata PROPERTIES PUBLIC_HEADER "${metadata_public_headers}")
 
-add_library(monoapi_mini INTERFACE)
+add_library(monoapi_jit INTERFACE)
 
-set(mini_public_headers_base
+set(jit_public_headers_base
   jit.h
-  mono-private-unstable.h)
-addprefix(mini_public_headers ./mono/mini "${mini_public_headers_base}")
+  mono-private-unstable.h
+  details/jit-types.h
+  details/jit-functions.h
+  details/mono-private-unstable-types.h
+  details/mono-private-unstable-functions.h
+  )
+addprefix(jit_public_headers ./mono/jit "${jit_public_headers_base}")
 
-set_target_properties(monoapi_mini PROPERTIES PUBLIC_HEADER "${mini_public_headers}")
+set_target_properties(monoapi_jit PROPERTIES PUBLIC_HEADER "${jit_public_headers}")
 
 add_library(monoapi INTERFACE)
-target_link_libraries(monoapi INTERFACE monoapi_utils monoapi_metadata monoapi_mini)
+target_link_libraries(monoapi INTERFACE monoapi_utils monoapi_metadata monoapi_jit)
 
 if(INSTALL_MONO_API)
-  install(TARGETS monoapi_utils PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/mono-2.0/mono/utils)
-  install(TARGETS monoapi_metadata PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/mono-2.0/mono/metadata)
-  install(TARGETS monoapi_mini PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/mono-2.0/mono/jit)
+  install(DIRECTORY mono/utils/ DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/mono-2.0/mono/utils)
+  install(DIRECTORY mono/metadata/ DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/mono-2.0/mono/metadata)
+  install(DIRECTORY mono/jit/ DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/mono-2.0/mono/jit)
```

Comment:
this will inadvertently install other files we put into these directories, e.g. README.md, right?
could we instead split `jit_public_headers_base` into the normal and `details/` .h files and `install` them separately here?

Reply:
> this will inadvertently install other files we put into these directories, e.g. README.md, right? could we instead split `jit_public_headers_base` etc. into the normal and `details/` .h files and `install` them separately here?

Yea I think so. I would just use `install(FILES)` then, I think, right?  I don't really know what i'm doing with cmake install rules.

Reply:
Yes that sounds good.

Reply:
We normally don't use `install(DIRECTORY`, @lambdageek. [Elsewhere](https://grep.app/search?q=install%28FILES&filter[repo.pattern][0]=dotnet/runtime&filter[lang][0]=CMake), we explicitly capture all the needed files in a variable, then use: `install(FILES ${that_var} DESTINATION path/to/some/dir)`.